### ISSUE TITLE
Swagger 2.0 Security Definitions (ex: oauth2)

### ISSUE THREAD

**Issue by wireframe** [fotinakis/swagger-blocks#11]
updating to the latest swagger-blocks gem and attempting to update my app from swagger 1.2 to 2.0 spec, this functionality appears to be missing.
https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#securityDefinitionsObject

**Comment by fotinakis** [comment]
It's there, it's just singular `security_definition` because you can specify different types of security definitions (same with some other similar types in swagger-blocks):

```Ruby
security_definition :api_key do
  key :type, 'apiKey'
  # ...
end
```

Let me know if that doesn't work! I'll leave this open to remind me to add something to the spec reference so it's more easily discoverable.

**Comment by wireframe** [comment]
Wow, that is super hidden.  I've been using the 2.0 example spec as some form of documentation and totally missed that feature.
https://github.com/fotinakis/swagger-blocks/blob/master/spec/lib/swagger_v2_blocks_spec.rb

thanks for replying and keep up the good work!

**Comment by avitus** [comment]
It would be great to have a complete example in the documentation. I also have looked repeatedly and didn't realize Swagger Blocks supported Oauth2.

**Comment by rhazegh** [comment]
+1 for example in the documentation.

**Comment by rhazegh** [comment]
@fotinakis this is how my `swagger_root` looks like in the `ApidocsController`.

```ruby
swagger_root do
  key :swagger, '2.0'
  info do
    key :version, '0.1.0'
    key :title, 'My API'
    key :description, 'My API description'
    key :termsOfService, 'http://localhost/terms/'
    contact do
      key :name, 'My Team'
    end
    license do
      key :name, 'Not assigned yet.'
    end
  end
  tags do
    key :name, 'food'
    key :description, 'Foods operations'
    externalDocs do
      key :description, 'Find more info here'
      key :url, 'http://website.example.com'
    end
  end
  key :host, 'localhost:3000'
  key :basePath, '/api/v1'
  key :consumes, ['application/json']
  key :produces, ['application/json']
  
  security_definition :my_oauth do
    key :type, 'oauth2'
    key :flow, 'implicit'
    key :authorizationUrl, 'http://localhost:3000/signin'
    key :scopes, 'public'  # <---- NOT SURE HOW TO DO THIS?
  end
end
```

I am using [doorkeeper] (https://github.com/doorkeeper-gem/doorkeeper#access-token-scopes) with the default `public` scope. 

- Can you please tell me how I should configure the last line `key :scopes, 'public'`?

- And, do you know if `Swagger-UI` supports the `implicit` flow for `swagger 2.0`?

**Comment by okolehao** [comment]
:+1:  it would be great to get an oauth2 example in there.

**Comment by fotinakis** [comment]
Hey everybody, I've published swagger-blocks 1.1.2 which fixes a small convention problem with security definition objects, and adds docs for how to support this.

Please update to 1.1.2 and check out the new docs:
https://github.com/fotinakis/swagger-blocks#security-handling

Let me know if this works!
